### PR TITLE
fix: transmit data from spring cloud config server controller to getEncryptor

### PR DESCRIPTION
**Pull request by RouxAntoine**
attention point to methods

https://github.com/spring-cloud/spring-cloud-config/blob/main/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java#L93

https://github.com/spring-cloud/spring-cloud-config/blob/main/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java#L84

which always have same bug but could not be properly fixed due to missing data in Get request

Fixes gh-2252

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public String encrypt(@RequestBody String data, @RequestHeader("Content-Type") M
 	@PostMapping("/encrypt/{name}/{profiles}")
 	public String encrypt(@PathVariable String name, @PathVariable String profiles, @RequestBody String data,
 			@RequestHeader("Content-Type") MediaType type) {
-		TextEncryptor encryptor = getEncryptor(name, profiles, "");
+		TextEncryptor encryptor = getEncryptor(name, profiles, data);
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java:106]: Shouldn't we be passing `input` instead?  Thats what the original code used

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java [hunk 2]
```diff
@@ -123,7 +123,7 @@ public String decrypt(@RequestBody String data, @RequestHeader("Content-Type") M
 	@PostMapping("/decrypt/{name}/{profiles}")
 	public String decrypt(@PathVariable String name, @PathVariable String profiles, @RequestBody String data,
 			@RequestHeader("Content-Type") MediaType type) {
-		TextEncryptor encryptor = getEncryptor(name, profiles, "");
+		TextEncryptor encryptor = getEncryptor(name, profiles, data);
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java:126]: We don't even need this line, `encryptor` is not used in the `catch` block so we can just move the variable definition into the `try` block which already passes `data`.

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
Could you please add tests?

**Comment by ryanjbaxter** [comment]
Ping @RouxAntoine see feedback

**Comment by RouxAntoine** [comment]
> Ping @RouxAntoine see feedback

hello, sorry for delay not so many time on my part I take a look

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java (hunk 1 above)
Comment (RouxAntoine) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java]: done in commit e7669e1b and use retrieved encryptor to encrypt message here https://github.com/spring-cloud/spring-cloud-config/pull/2253/files#diff-f18a83930fb437413479ed50adc18dedbf115b37e560d8481185a796c24444d8R111

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java (hunk 2 above)
Comment (RouxAntoine) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java]: done in commit e7669e1b

**Comment by RouxAntoine** [comment]
I just added two assert in a existing test. To check the key in encryptorLocator

**Pull request merged**